README.md
**ARCHIVE: The latest update is from 2014 and I have not touched it since :)**

BroadwayAPI (PHP)
=====================================
---- 
About
---- 

> Simple way to extract / export your PCTV Broadway HD-S2 channel playlist and current EPG using PHP.
> These can be reused in OpenELEC/ XBMC / Kodi in combination with the IPTV Simple Client

Usage
---- 

> **Prerequisites**:
> 
> - A PCTV Broadway HD-S2 box exists in your local network, is already setup and running :)
> - A server with PHP and access to your local network. 
> - Remote setups should work as well, as long as your Broadway setup is exposed to the internet (Router / DDNS Setup)
> 
> **Installation & Setup**:
> 
> See example. A more detailed example will follow.

Links
---- 

> OpenELEC          
>   - http://openelec.tv/
>    
> XBMC              
>   - http://xbmc.org/
> 	
> IPTV Simple Client    
>   - http://wiki.xbmc.org/index.php?title=Add-on:IPTV\_Simple\_Client

License
---- 

This is licensed under the [MIT License][1].

[1]:	/LICENSE.txt

changelog.txt
Version 1.0.0

- Initial release:
  - Export Playlist
  - Export EPG

resources/js/main.js
$( document ).ready(function() {
 /*
  setInterval(function(){
    $.ajax({ url: "overview.php", type: 'post',
        data: {'action': 'check_stream'}, success: function(data){
      
	   if(data.streamAvailable == true){
		   	$("#streamStatus").addClass("btn-default").removeClass("btn-danger").text('<span class="glyphicon glyphicon glyphicon-ok"></span> Stream available'+data.streamAvailable);
	   }else{
	   	$("#streamStatus").addClass("btn-danger").removeClass("btn-default").text('<span class="glyphicon glyphicon glyphicon-remove"></span> Stream in use'+data.streamAvailable);
	   }
        
    }, dataType: "json"});
}, 5000);
*/
$( ".sortable" ).sortable({ items: "tr", placeholder: "info" });

var delay = (function(){
  var timer = 0;
  return function(callback, ms){
    clearTimeout (timer);
    timer = setTimeout(callback, ms);
  };
})();
$("#configForm").on("keyup",function(){
	 var sdata =$("#configForm").serialize() 
	 delay(function(){
	 $(".spinner").show();
	 $.ajax({
  		url: "overview.php",
		type: 'post',
        data: sdata+"&action=save_config"
	  }).done(function(data,status) {
 		if(data.done == 1) $(".spinner").hide();
	  });}
	  ,2000);
	  
	 });

  $("#logoForm").on("keyup",function(){
	 var sdata =$("#logoForm").serialize() 
	 delay(function(){
	 $(".spinner").show();
	 $.ajax({
  		url: "overview.php",
		type: 'post',
        data: sdata+"&action=save_rename"
	  }).done(function(data,status) {
 		if(data.done == 1) $(".spinner").hide();
	  });}
	  ,2000);
	  
	 });
  $(".updateLocalFiles").click(function(){
	  $(".spinner").show();
	  $.ajax({
  		url: "overview.php",
		type: 'post',
        data: {'action': 'build'}
	  }).done(function(data,status) {
 		if(data.done == 1) $(".spinner").hide();
	  });
  })
 
 $('.navbar-nav li').tooltip();
 
});

class.BroadwayAPI.php
<?php
class BroadwayAPI{
		# Broadway network IP
		public $stream_ip		= "192.168.1.46";
		public $user 			= 'User';
		public $user_pin 		= '0000';
		
		# http://+IP+/TVC/user/data/profiles/m2ts
		# http://+IP+/TVC/user/data/profiles/flv
				     
		public $stream_profile 	= "";
		public $stream_caching	= 800;
		
		#Broadway channellist to use
		public $channel_list	= 1;
				
		static $channels;
		public $playlist;
		public $epg;
		static $channelEPG = array();	
		
		public $rename;
		public $config;			
		
		function __construct() {
			
			if(file_exists("resources/config.php")){
				
				$config = include("resources/config.php");
				$this->config = $config;
				
				$this->stream_ip 		= $config["stream_ip"];
				$this->stream_profile 	= $config["stream_profile"];
				$this->stream_caching 	= $config["stream_caching"];
				$this->channel_list 	= $config["channel_list"];	
				$this->user_pin 		= $config["user_pin"];		
			}
			
			if(file_exists("resources/rename.php")){
				$this->rename = include("resources/rename.php");
			}
		}
		
		/*
			Load Config
		*/
		function getConfig(){
			return $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/config");
		}
		function getTuners(){			
			return $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/free/data/tuners/1");	
		}
		/*
			Load Broadway profiles
		*/
		function getStreamProfiles(){
			
			$m2ts 	=  $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/profiles/m2ts");
			$flv 	=  $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/profiles/flv");
			
			return array('m2ts' => $m2ts,
						 'flv'  => $flv);	
		}
		
		/*
			Load Broadway channel listing / JSON
		*/
		function getChannelListing(){
			$listing =  $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/tv/channellists/");	
			
			foreach($listing as $list){
			
				$list->Items =  $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/tv/channellists/".$list->Id);	
				$data[] = $list;	
			}
			return $data;
		}
		
		/*
			Load Broadway Channellist / JSON
		*/
		function getChannelList(){
			return $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/tv/channellists/".$this->channel_list);	
		}
		
		/*
			Load Broadway Channel EPG / JSON
		*/

		function getChannelEPG($id){
			if ($this->channelEPG[$id] !== NULL  )
        		return $this->channelEPG[$id];
			
			$data = $this->getData("http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/epg/?ids=" .$id.  "&extended=1");
			
			$this->channelEPG[$id] = $data;
			
			return $data;	
		}
		
		/*
			Cleanup and combine channel list / epg
		*/
		function getChannels($addEPG = true){
			
			if ($this->channels !== NULL && $addEPG != true  )
        		return $this->channels;
		
			$data = array();
			
			$list = $this->getChannelList();
			
			foreach((array)$list->Channels as $channel){
				if($addEPG === true){
					$prep 			= $this->getChannelEPG($channel->Id);
	 				$entries		= $prep[0]->Entries;
	 				$channel->EPG 	= $entries;
				}
				
				$data[] 		= $channel;
			}
			
			$this->channels =  $data;
			return 	 $data;
		}

		/*
			Build Playlist
			- Images use the name of the channel with empty spaces or slashes replaced width "-"
			- In XBMC the image folder is defined in the IPTV Simple PVR Addon
		*/
		function buildPlaylist(){
			
			 $data = $this->getChannels(false);
			 
			 $playlist = "#EXTM3U\n";

                foreach($data as $d ){
                   
                        $playlist .="#EXTINF:-1 tvg-logo=\"".$this->cleanString($d->DisplayName) .".png\", ".$this->renameChannel($d->DisplayName) ."\n";
                        $playlist .=  "http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/basicauth/TVC/Preview?channel=".$d->Id . "&profile=".$this->stream_profile."&caching=".$this->stream_caching."\n";
				}
				
			// Add radio in the future
			/*	$playlist .="#EXTINF:-1 radio=\"true\", NDR 2\n";
				$playlist .="http://ndr-ndr2-nds-mp3.akacast.akamaistream.net/7/400/252763/v1/gnl.akacast.akamaistream.net/ndr_ndr2_nds_mp3";
			*/
			$this->playlist = $playlist;
		}

		/*
			Export Playlist
		*/
		function exportPlaylist($filename = "broadway.m3u"){
			
			$this->buildPlaylist();
			
			file_put_contents($filename, $this->playlist);	
		}
		
		/*
			Build EPG
			- Improvising on the category, as the Broadway EPG does not provide any further information			
		*/
		function buildEPG(){
			
			$data = $this->getChannels(true);
			
			$epg ="<?xml version=\"1.0\" encoding=\"UTF-8\" ?>\n";
            $epg .="<tv>\n";

               foreach($data as $d ){
                    $epg .="<channel id=\"".$d->Id . "\">\n";
                    $epg .=" <display-name>".$this->renameChannel($d->DisplayName) ."</display-name>\n";				
                    $epg .="</channel>\n";
			   }
			
			   foreach($data as $d ){
					
					 foreach($d->EPG as $prog ){
			            if(preg_match("/,/", $prog->ShortDescription, $matches)){
							$split = explode(",",  $prog->ShortDescription);
							if(count($split) == 3){
									$subtitle = $split[0]." / ".$split[2];
									$cat = $split[1];
							}else{
									$subtitle = $split[1];
									$cat = $split[0];
							}
							
							if(empty($cat)){
								$cat = "News";	
							}
							
							$epg .="<programme start=\"".date("YmdHis",($prog->StartTime/1000))." +0200\" stop=\"".date("YmdHis",($prog->EndTime/1000))." +0200\" channel=\"".$d->Id."\">\n";                           
							$epg .=" <title>".$this->xml_entities($prog->Title)." (".$this->xml_entities($subtitle).")</title>\n";
							$epg .=" <desc>".$this->xml_entities($prog->LongDescription)."</desc>\n";
							$epg .=" <category>".$this->xml_entities($cat) ."</category>\n";
							$epg .="</programme>\n";	
						}else{ 
					   		$epg .="<programme start=\"".date("YmdHis",($prog->StartTime/1000))." +0200\" stop=\"".date("YmdHis",($prog->EndTime/1000))." +0200\" channel=\"".$d->Id."\">\n";                           
					   		$epg .=" <title>".$this->xml_entities($prog->Title)."</title>\n";
					   		$epg .=" <desc>".$this->xml_entities($prog->LongDescriptio)."</desc>\n";
					   		$epg .=" <category>".$this->xml_entities($prog->ShortDescription)."</category>\n";
					   		$epg .="</programme>\n";
					 	}
					 }
				}
 			$epg .="</tv>\n";
			
 			$this->epg =  $epg;
		}
		
		function renameChannel($channel){
			if(!empty($this->rename[$channel])){
				return 	trim($this->rename[$channel]);
			}	
			return $channel;
		}
		/*
			Export EPG
		*/
		function exportEPG($filename = "broadway_epg.xml"){
			
			$this->buildEPG();
			
			file_put_contents($filename, $this->epg);	
		}
		
		function checkForBroadway(){		
			
			$url =  "http://".$this->user.":".md5($this->user_pin)."@".$this->stream_ip."/TVC/user/data/tv/channellists"; 
						
			switch($this->getResponseCode($url)){
				case 200:
					return true;
				break;
				
			}	
			return false;
			
		}
		/* Check, if stream is occupied */
		
		function isStreamAvailable(){
			
			$tuner = $this->getTuners();
			if($tuner->Status->TuningStatus != "Tuned") return true;
			return false;
		}
		
		function getResponseCode($url) {
    		$headers = get_headers($url);
    		return substr($headers[0], 9, 3);
		}
		
		function getRedirectsToUri($uri)
		{
			$redirects = array();
			$http = stream_context_create();
			stream_context_set_params(
				$http,
				array(
					"notification" => function() use (&$redirects)
					{
						if (func_get_arg(0) === STREAM_NOTIFY_REDIRECTED) {
							$redirects[] = func_get_arg(2);
						}
					}
				)
			);
			file_get_contents($uri, false, $http);
			return $redirects;
		}
		
		function cleanString($string){
  			 $upas = Array("ä" => "ae", "ü" => "ue", "ö" => "oe", "Ä" => "Ae", "Ü" => "Ue", "Ö" => "Oe", "/"=> "-"," "=> "-"); 
 			 return  strtr($string, $upas);
  }
		
		/*
			Fetch data
		*/
		function getData($url){
			
			$ch = curl_init();

			curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
			
			curl_setopt($ch, CURLOPT_HEADER, 0);
			curl_setopt($ch, CURLOPT_USERAGENT, $_SERVER['HTTP_USER_AGENT']);
			curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
			
			curl_setopt($ch, CURLOPT_URL, $url);
			
			$result=curl_exec($ch);
		
			curl_close($ch);		
			
			return json_decode($result);
		}
		
		function xml_entities($string) {
		  return strtr(
			  $string, 
			  array(
				  "<" => "&lt;",
				  ">" => "&gt;",
				  '"' => "&quot;",
				  "'" => "&apos;",
				  "&" => "&amp;",
			  )
		  );
	}
}

overview.php
<?php

include("class.BroadwayAPI.php");

$Broadway = new BroadwayAPI();

$listings 			= $Broadway->getChannelListing();
$config 			= $Broadway->config;
$rename 			= $Broadway->rename;

$stream_profiles 	= $Broadway->getStreamProfiles();

$streamAvailable 	= $Broadway->isStreamAvailable();
$broadwayAvailable 	= $Broadway->checkForBroadway();

if(!empty($_POST['action'])){
	
	switch($_POST['action']){
	case "save_rename":
		header("content-type:application/json");
		$ini = "<?php\n #Rename channels\n\n return array(\n\n";
		
		foreach($_POST['rename'] as $key => $val){
				if(!empty($val)) $ini .=	"'".utf8_decode($key)."' => '".$val."',\n";
		
		}
		$ini .= "\n\n);";
		file_put_contents("resources/rename.php", $ini);
		$return['done'] = 1;
		$return["json"] = json_encode($return);
		echo json_encode($return);
		exit;
	break;
	case "save_config":
		header("content-type:application/json");
		
		$ini = "<?php\n #BroadwayAPI configuration\n\n return array(\n\n";
		
		foreach($_POST['config'] as $key => $val){
				 $ini .=	"'".utf8_decode($key)."' => '".$val."',\n";
		
		}
		$ini .= "\n\n);";
		
		file_put_contents("resources/config.php", $ini);
		
		$return['done'] = 1;
		$return["json"] = json_encode($return);
		echo json_encode($return);
		exit;
	break;
	case "build":
		header("content-type:application/json");
		
		// Save playlist to file
		$Broadway->exportPlaylist("broadway.m3u");
	
		// Save EPG data to file
		$Broadway->exportEPG("broadway_epg.xml");
		
		$return['done'] = 1;
		$return["json"] = json_encode($return);
		echo json_encode($return);
		exit;
	break;
	case "check_stream":
		header("content-type:application/json");		
		$return['streamAvailable'] = $Broadway->isStreamAvailable();
		$return["json"] = json_encode($return);
		echo json_encode($return);
		exit;
	
	break;
	
	}
}


?>
<!DOCTYPE html>
<html lang="en">
<head>
<title>BroadwayAPI (PHP)</title>
<link rel="stylesheet" href="//maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css">
<link rel="stylesheet" href="/resources/css/bootstrap.min.css">
<link rel="stylesheet" href="//code.jquery.com/ui/1.11.1/themes/smoothness/jquery-ui.css">
<link rel="stylesheet" href="/resources/css/main.css">
<script src="//code.jquery.com/jquery-1.11.0.min.js"></script>
<script src="//code.jquery.com/jquery-migrate-1.2.1.min.js"></script>
 <script src="//code.jquery.com/ui/1.11.1/jquery-ui.js"></script>
<script src="//maxcdn.bootstrapcdn.com/bootstrap/3.2.0/js/bootstrap.min.js"></script>
<script src="/resources/js/main.js"></script>
</head>
<body><br>
<div class="container"> <img src="/resources/images/logo.png" width="400">
  <div class="navbar navbar-default" role="navigation">
    <div class="container-fluid">
   
      <div class="navbar-header">
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span> </button>
       </div>
      <div class="navbar-collapse collapse">
        <ul class="nav navbar-nav">
      <li class="dropdown">
    <a class="dropdown-toggle" data-toggle="dropdown" href="#">
      Broadway <span class="caret"></span>
    </a>
    <ul class="dropdown-menu" role="menu">
    <li><a href="http://<?php echo $Broadway->stream_ip; ?>" target="_blank">LiveTV</a></li>
          <li ><a href="http://<?php echo $Broadway->stream_ip; ?>/TVC.1343/ui/broadway/Admin.html" target="_blank">Admin</a></li>
          <li ><a href="http://<?php echo $Broadway->stream_ip; ?>/TVC.1343/ui/Settings.html" target="_blank">Settings</a></li>
    </ul>
  </li>
          
          <li data-toggle="tooltip" data-placement="bottom" title="Update Playlist & EPG files"><a href="#" class="updateLocalFiles">Update Files</a></li>
        </ul>
        <ul class="nav navbar-nav navbar-right">
          <li ><a href="https://bitbucket.org/portalzine/cubicfusion-broadwayapi" target="_blank">Repository</a></li>
        </ul>
      </div>
    </div>
  </div>
  <div class="spinner">
    <div class="rect1"></div>
    <div class="rect2"></div>
    <div class="rect3"></div>
    <div class="rect4"></div>
    <div class="rect5"></div>
  </div>
  <ul class="nav nav-tabs" role="tablist">
    <li class="active" ><a href="#home" role="tab" data-toggle="tab">Home</a></li>
    <li><a href="#setting" role="tab" data-toggle="tab">Settings</a></li>
    <li><a href="#list" role="tab" data-toggle="tab">Channel Lists</a></li>
    <li><a href="#logos" role="tab" data-toggle="tab">Channels & Logos</a></li>
    <li><a href="#add" role="tab" data-toggle="tab">Add Streams</a></li>
  </ul>
  <div class="tab-content">
  <div class="tab-pane " id="add">
      <div class="jumbotron">
     <h2>Add Stream</h2>
     <small>Add radio and video streams to your playlist.</small><br><br>
      <form role="form" method="POST" action="">
  <div class="form-group">
    <label for="name">Name</label>
    <input type="text" class="form-control" id="name" name="add[name]" placeholder="">
  </div>
   <div class="form-group">
    <label for="name">Stream</label>
    <input type="text" class="form-control" id="stream" name="add[stream]" placeholder="">
  </div>
   <div class="form-group">
    <label for="name">Logo</label>
    <input type="text" class="form-control" id="logo" name="add[logo]" placeholder="">
  </div>
  <div class="checkbox-inline">
    <label>
      <input type="radio" name="add[type]" value="video"> Video 
    </label>
  </div>
   <div class="checkbox-inline" >
    <label>
      <input type="radio" name="add[type]" value="radio"> Radio
    </label>
  </div>
  <div class="clearfix"></div><br><br>
  <button type="submit" class="btn btn-default btn-block">Submit</button>
</form>
<br><hr ><br>
 <h2>Video</h2>
        <table class="table table-striped sortable" >
         <tr>
              <th>Name</th>
              <th>Stream</th>
              <th>Logo</th>
            </tr>
           
        </table><br>
        <h2>Radio</h2>
        <table class="table table-striped sortable">
         <tr>
              <th>Name</th>
              <th>Stream</th>
              <th>Logo</th>
            </tr>
          
        </table>
      </div>
    </div>
    <div class="tab-pane active" id="home">
      <div class="jumbotron">
        <h2>Status</h2>
        <div class="btn-group"> <button id="networkStatus" type="button" <?php if($broadwayAvailable){echo 'class="btn btn-default"><span class="glyphicon glyphicon glyphicon-ok"></span> Broadway live';}else{ echo 'class="btn-danger"><span class="glyphicon glyphicon glyphicon-remove"></span>Broadway not live' ;} ?>
          </button>
          <button type="button" id="streamStatus" <?php if($streamAvailable){echo 'class="btn btn-default"><span class="glyphicon glyphicon glyphicon-ok"></span> Stream available';}else{echo 'class="btn btn-danger"><span class="glyphicon glyphicon glyphicon-remove"></span> Stream in use' ;}  ?> 
          </button>
        </div>
        <br><br>
         <h2>Exports</h2>
        <table class="table table-striped">
         <tr>
              <td>Playlist (m3u)<br>
                <span class="label label-primary"><?php echo date("d.m.Y H:i:s.", filectime("broadway.m3u")); ?></span></td>
              <td><input class='form-control' type='text' value='http://<?php echo $_SERVER["SERVER_NAME"]; ?>/broadway.m3u'></td>
            </tr>
            <tr>
              <td>XMLTV<br>
                <span class="label label-primary"> <?php echo date("d.m.Y H:i:s.", filectime("broadway_epg.xml")); ?></span></td>
              <td><input class='form-control' type='text' value='http://<?php echo $_SERVER["SERVER_NAME"]; ?>/broadway_epg.xml'></td>
            </tr>
        </table>
         <br><br>
         <h2>Stream-Profiles m2ts</h2>
         <table class="table table-striped">
          <thead>
          <tr>
              <th>ID</th>
               <th>Video Bitrate</th>
               <th>Audio Bitrate</th>
              <th>Dimension</th>
            </tr>
            </thead>
		 <?php foreach($stream_profiles['m2ts'] as $m2ts){?>
         <tr>
              <td><?php echo $m2ts->Id; ?></td>
               <td><?php echo $m2ts->{'Video.Bitrate'}; ?></td>
               <td><?php echo $m2ts->{'Audio.Bitrate'}; ?></td>
              <td><?php echo $m2ts->{'Width'}; ?>x<?php echo $m2ts->{'Height'}; ?></td>
            </tr>
           <?php }?>
        </table>
        <br><br>
         <h2>Stream-Profiles flv</h2>
         <table class="table table-striped">
          <thead>
          <tr>
              <th>ID</th>
               <th>Video Bitrate</th>
               <th>Audio Bitrate</th>
              <th>Dimension</th>
            </tr>
            </thead>
		 <?php foreach($stream_profiles['flv'] as $m2ts){?>
         <tr>
              <td><?php echo $m2ts->Id; ?></td>
               <td><?php echo $m2ts->{'Video.Bitrate'}; ?></td>
               <td><?php echo $m2ts->{'Audio.Bitrate'}; ?></td>
              <td><?php echo $m2ts->{'Width'}; ?>x<?php echo $m2ts->{'Height'}; ?></td>
            </tr>
           <?php }?>
        </table>
      </div>
    </div>
    <div class="tab-pane" id="setting">
      <div class="jumbotron">
        <h2>Settings</h2>
        <form role="form" id="configForm">
          <table class="table table-striped">
            <?php
	
		
	  foreach($config as $key => $var){
		  if($key == "user_pin"){
			  echo "<tr>
		<td>TV-PIN <br>
<small>default: 0000</small></td>
		<td><input class='form-control' type='password' value='0000' name='config[".$key."]'></td>		
		</tr>";
			  }elseif($key =="channel_list"){
				 
		
		echo "<tr>
		<td>".$key."</td><td><select class='form-control' name='config[".$key."]'><option value=''>Choose</option>";
		 foreach($listings as $list){
			 echo "<option value='".$list->Id."'";
			 if($var ==$list->Id) echo " selected";
			 echo">".$list->DisplayName."</option>";
		}
		echo"</select></td>		
		</tr>";
		
				  }
				  elseif($key =="stream_profile"){
				 
		
		echo "<tr>
		<td>".$key."</td><td><select class='form-control' name='config[".$key."]'><option value=''>Choose</option>";
		foreach($stream_profiles['m2ts'] as $m2ts){
			 echo "<option value='".$m2ts->Id."'";
			 if($var ==$m2ts->Id) echo " selected";
			 echo">".$m2ts->Id." / ".$m2ts->{'Video.Bitrate'}."</option>";
		}
		
		echo"</select></td>		
		</tr>";
		
				  }
			  
			  else{
echo "<tr>
		<td>".$key."</td>
		<td><input class='form-control' type='text' value='".$var."' name='config[".$key."]'></td>		
		</tr>";
			  }
	  }
		?>
           
          </table>
        </form>
      </div>
    </div>
    <div class="tab-pane" id="list">
      <div class="jumbotron">
        <h2>Channellists</h2>
        <table class="table table-striped">
          <?php
echo "<thead><tr>

		<th>ID</td>
		<th>NAME</td>
		<th>CHANNELS</td>
		
		</tr></thead><tbody>";
		
foreach($listings as $list){
		echo "<tr class='success'>
		<td>".$list->Id."</td>
		<td>".$list->DisplayName."</td>
		<td>".$list->Count."</td>
		</tr>";
		
		
		echo "<tr>
		
		<td colspan='3'>";
		foreach($list->Items->Channels as $channel){
			echo '<span class="label label-primary">'.$channel->DisplayName.'</span> ';
		}
		echo "</td>
		
		</tr></tbody>";
}
?>
        </table>
      </div>
    </div>
    <div class="tab-pane" id="logos">
      <div class="jumbotron">
        <h2>Channel Logos</h2> <form id="logoForm">
        <table class="table table-striped">
           
          
          <?php
echo "<thead><tr >

		<th>ID</td>
		<th>NAME</td>
		<th>CHANNELS</td>
		
		</tr></thead><tbody>";
		
foreach($listings as $list){
		echo "<tr class='success'>
		<td>".$list->Id."</td>
		<td>".$list->DisplayName."</td>
		<td>".$list->Count."</td>
		</tr>";
		
		
		
		foreach($list->Items->Channels as $channel){echo "<tr>
		
		<td colspan='2'>";
			echo '<h4>'.utf8_decode($channel->DisplayName).'</h4>
<small>[ID: '.$channel->Id.' / '.$Broadway->cleanString($channel->DisplayName) .'.png]</small><br>
<input class="form-control" type="text" name="rename['.$channel->DisplayName.']" value="'.$rename[$channel->DisplayName].'" placeholder="Rename"></td>';
			if(file_exists($config['channel_logos'].$Broadway->cleanString($channel->DisplayName) .".png")){
			echo 	"<td class='active'><center><img width='80' src='/".$config['channel_logos'].$Broadway->cleanString($channel->DisplayName) .".png'></center></td>";
			}else{
				echo 	"<td class='danger'>Missing: ".$Broadway->cleanString($channel->DisplayName) .".png</td>";
				}
		echo "</td></tr>";
		}
		
	
		
		echo "</tbody>";
}
?>
        </table>
        </form>
      </div>
    </div>
  </div>
  <center><img src="/resources/images/tai.jpg">
    <small>&copy; Copyright 2014 <a href="http://www.portalzine.de" target="_blank">portalZINE NMN</a> / Alexander Graef. All rights reserved.</small>
  </center>
  <br>
  <br>
</div>

</body>
</html>
